A Cypress test failed:

> isValid
>        should showing valid correctly with schema validation and default value supplied:
>      AssertionError: Timed out retrying: Expected to find content: 'false' within the element: <div#isValid> but never did.
>       at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/isValid.ts:146:28)

https://github.com/react-hook-form/react-hook-form/blob/master/cypress/integration/isValid.ts
```
  it('should showing valid correctly with schema validation and default value supplied', () => {
    cy.visit('http://localhost:3000/isValid/schema/defaultValues');
    cy.get('#isValid').contains('true');

    cy.get('input[name="firstName"]').clear();
    cy.get('#isValid').contains('false');
    cy.get('#renderCount').contains('2');
    cy.get('input[name="firstName"]').type('test');
    cy.get('#isValid').contains('true');
    cy.get('#isValid').contains('true');
    cy.get('#toggle').click();
    cy.get('#isValid').contains('false');
    cy.get('#toggle').click();
    cy.get('input[name="firstName"]').type('t');
    cy.get('#isValid').contains('true');
  });
```
This is the test page: https://github.com/react-hook-form/react-hook-form/blob/master/app/src/isValid.tsx

Not sure exactly which assertion failed, but spontaneously I think that second last line `type('t')` shouldn't be necessary to get `isValid==true`. Toggling a valid fields visibility off and on should make the form valid without having to modify the input no?

--- a/src/useForm.ts
+++ b/src/useForm.ts
@@ -701,12 +701,14 @@ export function useForm<
 
   const validateResolver = React.useCallback(
     async (values = {}) => {
+      const newDefaultValues = isEmptyObject(fieldsRef.current)
+        ? defaultValuesRef.current
+        : {};
+
       const { errors } =
         (await resolverRef.current!(
           {
-            ...(isEmptyObject(fieldsRef.current)
-              ? { ...defaultValuesRef.current }
-              : {}),
+            ...newDefaultValues,
             ...getValues(),
             ...values,
           },

--- a/src/useForm.ts
+++ b/src/useForm.ts
@@ -704,7 +704,9 @@ export function useForm<
       const { errors } =
         (await resolverRef.current!(
           {
-            ...defaultValuesRef.current,
+            ...(isEmptyObject(fieldsRef.current)
+              ? { ...defaultValuesRef.current }
+              : {}),
             ...getValues(),
             ...values,
           },

